Source/Particles/PhysicalParticleContainer.cpp

```diff
@@ -1977,6 +1979,25 @@ PhysicalParticleContainer::getIonizationFunc (const WarpXParIter& pti,
                                 ion_atomic_number);
 }
 
+void PhysicalParticleContainer::resample (Resampling& resampler, const int timestep)
```

Comment:
Would calling this method result in modifications of the `resampler` object? If this is not the case, you can use a `const` reference...

Reply:
Yes it would, because we initialize the number of cells in the `ResamplingTrigger` object the first time we call this method. But I guess this will be different if we use `mutable` as you suggested below.